### ISSUE TITLE
changefeed may stuck when tikv upgrade/restart/evict leader 

### ISSUE THREAD

**Issue by asddongmen** [pingcap/tiflow#10584]
### What did you do?

Rolling upgrade tikv.

### What did you expect to see?

The changefeed checkpointTs lag less than 30s.
And more than thousand `strean to store closed` logs was found in cdc logs within 10 seconds.

### What did you see instead?

The changefeed checkpointTs lag more than 10 minutes.
Additionally, more than a thousand `stream to store closed logs` were found in the cdc logs within 10 seconds.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
v6.5.7
```

**Comment by asddongmen** [comment]
ref: https://github.com/pingcap/tiflow/issues/10239#issuecomment-1914075973

**Comment by asddongmen** [comment]
fix by #10559
